vehicle.py

```diff
@@ -0,0 +1,262 @@
+        self.damage = gb_cf.DAMAGE[self.model.vehicle_type]
+        self.priority = None
+
+    def refresh_model(self, state: GameState) -> None:
+        """
+        Refreshes tank model using game state
+        :param state: GameState object
+        :return: None
+        """
+        self.model = state.our_tanks[self.t_id]
+
+    def cells_in_range(self) -> set[Coordinates]:
+        """
+        Returns cells in shooting range of tank
+        :return: set of Coordinates
+        """
+        max_radius = self.max_shoot_range + self.model.shoot_range_bonus
+        return self.model.coordinates.in_radius_excl(self.min_range, max_radius)
+
+    def targets_in_range(self, state: GameState, map_: GameMap
+                         ) -> Union[set, set[Coordinates]]:
+        """
+        Returns cells with non-neutral player
+        tanks that are in range of current tank
+        :param state: GameState obj
+        :param map_: GameMap obj
+        :return: set of cells
+        """
+        return self.cells_in_range().intersection(state.get_aggressive_cells())
+
+    def shoot(self, target: Coordinates) -> tuple[Actions, dict]:
+        """
+        Produces SHOOT action with self t_id and target coordinate
+        :param target: cell with enemy tank
+        :return: Python obj SHOOT action
+        """
+        return Actions.SHOOT, {"vehicle_id": self.t_id,
+                               "target": {"x": target[0], "y": target[1], "z": target[2]}}
+
+    def move(self, cell: Coordinates) -> tuple[Actions, dict]:
+        """
+        Produces MOVE action with self t_id and target coordinate
+        :param cell: empty Coordinate obj in moving range
+        :return: Pyhon obj MOVE action
+        """
+        return Actions.MOVE, {"vehicle_id": self.t_id,
+                              "target": {"x": cell[0], "y": cell[1], "z": cell[2]}}
+
+    def choose_target(self, targets: set[Coordinates], state: GameState) -> Coordinates:
+        """
+        Choose target from given set of available
+        targets, according to given GameState
+        :param targets: set of available targets
+        :param state: GameState obj
+        :return: target cell
+        """
+        for target in targets:
+            if state.aggressive_tanks[target] <= self.damage:
+                return target
+        return targets.pop()
+
+    def make_turn(self, state: GameState, map_: GameMap
+                  ) -> Optional[tuple[Actions, dict]]:
+        """
+        Main method to provide vehicle action, returns
+        Python obj with SHOOT or MOVE action | None
+        :param state: GameState obj
+        :param map_: MapState obj
+        :return: Python obj action
+        """
+        self.refresh_model(state)
+        targets = self.targets_in_range(state, map_)
+        if targets:
+            target = self.choose_target(targets, state)
+            return self.shoot(target)
+        self.set_priority(state, map_)
```

Comment:
It is a good practice to separate different logic blocks of code with a blank line.